### ISSUE TITLE
Having heterogeneous combinations of security handlers does not work properly

### ISSUE THREAD

**Issue by mohak2006** [cdimascio/express-openapi-validator#736]
If a particular endpoint spec contains multiple security handlers with both AND and OR conditions according to the "Using Multiple Authentication Types" section on this link: https://swagger.io/docs/specification/authentication/, then the behaviour of the API is not as expected. 

For e.g. take following scenario:
```
paths:
  /health_check:
    get:
      security:
        - Handler1ThatReturnTrue: []
          Handler2ThatReturnFalse: []
        - Handler3ThatReturnTrue: []
          Handler4ThatReturnTrue: []
```

The Above one will fail authorization.

If you change the above scenario to:
```
paths:
  /health_check:
    get:
      security:
        - Handler1ThatReturnTrue: []
          Handler2ThatReturnTrue: []
        - Handler3ThatReturnTrue: []
          Handler4ThatReturnFalse: []
```

This scenario will pass authorization.

Somehow when you use both AND and OR combinations of security handlers, then only the first combination is respected. Can you please check and provide a fix or a workaround for this issue?

**Comment by wolfpack94** [comment]
Is there going to be any fix for this?

**Comment by jm12131415** [comment]
I have the same issue too. Any updates on this?

**Comment by lukebellamy053** [comment]
I have an MR for fixing the issue where it fails auth when one of the schemas should have passed: https://github.com/cdimascio/express-openapi-validator/pull/946

**Comment by cdimascio** [comment]
Closing #946 is merged and available in 5.3.4
